lib/composer_messages_finder.rb

```diff
@@ -226,6 +227,24 @@ def check_get_a_room(min_users_posted: 5)
     }
   end
 
+  def check_dont_feed_the_trolls
+    return if @post.blank? || !replying?
+
+    flags = @post.flags.group(:user_id).count
```

Comment:
I wonder if we can be slightly more efficient here. It doesn't seem like we need to create a `Post` ActiveRecord object here and can simply query the `PostAction` table with the post's id. This means we save on creating an AR object and a DB query to find the post in most cases.

Reply:
My main concern would be that the knowledge of how to get a post's flags is encoded in `Post#flags`, and would need to be duplicated here, introducing a chance that these two sources of truths drift.

One middle of the road approach could be to just instantiate a `Post` with the appropriate ID, which would avoid the initial DB query, but still works for getting the flags (and returns a reasonable null value of empty hash if the post doesn't exist.) WDYT?

Reply:
> My main concern would be that the knowledge of how to get a post's flags is encoded in Post#flags

Perhaps a `Post.flags_for_post` class method which `Post#flags` calls? That would allow us to keep the logic of getting a post's flags in a single method.

> One middle of the road approach could be to just instantiate a Post with the appropriate ID, which would avoid the initial DB query, but still works for getting the flags (and returns a reasonable null value of empty hash if the post doesn't exist.) WDYT?

Small clarification here, if we're instantiating a `Post` we'll still be paying the cost of loading the post when we technically don't need to.